Library/Formula/ice.rb

```diff
@@ -42,17 +30,35 @@ def install
 
       # BerkeleyDB dislikes parallel builds
       ENV.deparallelize
-
-      # --enable-compat185 is necessary because our build shadows
-      # the system berkeley db 1.x
       args = %W[
         --disable-debug
         --prefix=#{libexec}
         --mandir=#{libexec}/man
         --enable-cxx
-        --enable-compat185
       ]
 
+      if build.with? "java"
+        java_home = ENV["JAVA_HOME"] = `/usr/libexec/java_home -v 1.7`.chomp
+
+        # The Oracle JDK puts jni.h into #{java_home}/include and jni_md.h into
+        # #{java_home}/include/darwin.  The original Apple/SUN JDK placed jni.h
+        # and jni_md.h into
+        # /System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers/
+        #
+        # Setup the include path with the Oracle JDK location first and the Apple JDK location second.
+        ENV.append("CPPFLAGS", "-I#{java_home}/include")
```

Comment:
Are these necessary? This should be being automatically passed by the `:java` dependency and safe to not need?